tests/testsuite.src/syn_misc.at

```diff
@@ -8348,3 +8348,253 @@ AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [1])
 
 AT_CLEANUP
 
+
+AT_SETUP([Check recursive calls])
+AT_KEYWORDS([RECURSIVE])
+
+# Some of these tests are expected to fail, either at compile time or run
```

Comment:
It seems this test should be split into at least two tests - one with an expected fail (coming up with some reasonable error messages which we can adjust later as we see fit) and one with expected pass.

The one that is run should be moved to run_misc.at, syn_misc.at should in nearly all cases only do `$COMPILE_ONLY` .

Reply:
So should I add an `AT_XFAIL` for those tests that *should* fail but actually pass (which means running the testsuite sill fail with "unexpected pass") ?